testsuite/domain/src/test/java/org/jboss/as/test/integration/domain/management/cli/DeployAllDomainTestCase.java

```diff
@@ -123,69 +113,505 @@ public static void after() throws Exception {
     public void beforeTest() throws Exception {
         ctx = CLITestUtil.getCommandContext(testSupport);
         ctx.connectController();
+        infoUtils.enableDoubleCheck(ctx);
+    }
+
+    @After
+    public void afterTest() {
+        ctx.handleSafe("deployment undeploy * --all-relevant-server-groups");
+        ctx.terminateSession();
+        infoUtils.resetDoubleCheck();
+    }
+
+    /**
+     * Test verify a live cycle of deployment operation in singleton mode with Aesh commands.
+     * <ul>
+     * <li>Step 1) Deploy applications deployments to defined server groups</li>
+     * <li>Step 2a) Verify if deployment are successful by list command</li>
+     * <li>Step 2b) Verify if applications deployments are enabled for defined server groups by info command</li>
+     * <li>Step 3a) Disabling two selected applications deployments</li>
+     * <li>Step 3b) Try disabling application deployment in wrong server group space. expect command execution fail</li>
+     * <li>Step 4) Verify if two selected applications deployments are disabled, but other have still previous state</li>
+     * <li>Step 5) Disable all deployed applications deployments in all server groups</li>
+     * <li>Step 6) Check if all applications deployments is disabled in all server groups</li>
+     * <li>Step 7) Enable all applications deployments for all server groups</li>
+     * <li>Step 8) Verify if all applications deployments are enabled for all server groups</li>
+     * <li>Step 9) Undeploy one application deployment</li>
+     * <li>Step 10) Check if selected application deployment is removed, but others still exist with right state</li>
+     * <li>Step 11) Undeploy all applications deployments</li>
+     * <li>Step 12) Check if all applications deployments is gone</li>
+     * </ul>
+     *
+     * @throws Exception
+     */
+    @Test
+    public void testDeploymentLiveCycleWithServerGroups() throws Exception {
+        // Step 1) Deploy applications deployments to defined server groups
+        ctx.handle("deployment deploy-file --server-groups=" + sgOne + ' ' + cliTestApp1War.getAbsolutePath());
+        ctx.handle("deployment deploy-file --server-groups=" + sgOne + ' ' + cliTestAnotherWar.getAbsolutePath());
+        ctx.handle("deployment deploy-file --server-groups=" + sgTwo + ' ' + cliTestApp2War.getAbsolutePath());
+        ctx.handle("deployment deploy-file --server-groups=" + sgTwo + ',' + sgOne + ' ' + cliTestAppEar.getAbsolutePath());
+
+        // Step 2a) Verify if deployment are successful by list command
```

Comment:
If I understand what is going on there, you first call .checkDeploymentByList that does the actual call and fill the infoUtils.rows . Then you just checkExistInOutputMemory that just checks the row. You can see that there is now an implicit order between the calls. This makes for difficult understanding. I think tha tthis highlight the benefit of having a stateless Utility class. What do you think of the following piece of code:
List<String> rows = InfoUtils.listDeployments(cli);
InfoUtils.checkExists(rows, cliTestApp1War.getName());
Your utility class would benefit from it I think.

Reply:
Thank you, you're right, I'll try to make some adjustments.

Reply:
Is it problem have internal public static class in Utils class?

Reply:
No, you can define a static Util class.